rust/kernel/file_operations.rs

```diff
@@ -84,26 +86,40 @@ unsafe extern "C" fn open_callback<A: FileOpenAdapter, T: FileOpener<A::Arg>>(
     file: *mut bindings::file,
 ) -> c_types::c_int {
     from_kernel_result! {
-        let arg = A::convert(inode, file);
+        let mut file = NonNull::new_unchecked(file);
```

Comment:
`NonNull` isn't very helpful if it's only used as local variable. The original code is easier to read.